.github/scripts/gitutils.py

```diff
@@ -211,22 +211,53 @@ def amend_commit_message(self, msg: str) -> None:
         self._run_git("commit", "--amend", "-m", msg)
 
 
-def parse_args() -> Any:
-    from argparse import ArgumentParser
-    parser = ArgumentParser("Merge PR/branch into default branch")
-    parser.add_argument("--sync-branch", default="sync")
-    parser.add_argument("--default-branch", type=str, default="main")
-    parser.add_argument("--dry-run", action="store_true")
-    return parser.parse_args()
+class PeekableIterator:
+    def __init__(self, val: str) -> None:
+        self._val = val
+        self._idx = -1
 
+    def peek(self) -> Optional[str]:
+        if self._idx + 1 >= len(self._val):
+            return None
+        return self._val[self._idx + 1]
 
-def main() -> None:
-    args = parse_args()
-    repo = GitRepo(get_git_repo_dir())
-    repo.cherry_pick_commits(args.sync_branch, args.default_branch)
-    if not args.dry_run:
-        repo.push(args.default_branch)
+    def __iter__(self) -> Any:
```

Comment:
IMO replace `Any` with `"PeekableIterator"` and make this class extend `collections.abc.Iterator[str]`

But I guess this is fine as-is